rotkehlchen/externalapis/gnosispay.py

```diff
@@ -181,10 +181,9 @@ def maybe_deserialize_transaction(self, data: dict[str, Any]) -> GnosisPayTransa
                 reversal_tx_hash=None,  # atm does not appear in the API
             )
 
-        except KeyError as e:
-            log.error(f'Could not find key {e!s} in Gnosis pay transaction response: {data}')
-        except DeserializationError as e:
-            log.error(f'Failed to read gnosis pay data {data} due to {e!s}')
+        except (DeserializationError, KeyError, IndexError) as e:
+            msg = f'missing key: {e}' if isinstance(e, KeyError) else str(e)
```

Comment:
no this is not the right approach. As you saw from the data I showed you there can be gnosis pay refunds without a specific transaction. That's a valid thing. You should not just blindly catch th eexception and assume that `data['transactions']` can not be empty.

The data I showed you should make that obvious.